src/phoenix/session/client.py

```diff
@@ -480,58 +548,108 @@ def create_examples(
+
+    def upload_dataset(
+        self,
+        table: Union[str, Path, pd.DataFrame],
+        /,
+        *,
+        dataset_name: str,
+        input_keys: Iterable[str],
+        output_keys: Iterable[str] = (),
+        metadata_keys: Iterable[str] = (),
+        dataset_description: Optional[str] = None,
+    ) -> Dataset:
+        """
+        Upload examples as dataset to the Phoenix server.
+
+        Args:
+            table (str | Path | pd.DataFrame): Location of a CSV text file, or
+                pandas DataFrame.
+            dataset_name: (str): Name of the dataset. Required if action=append.
+            input_keys (Iterable[str]): List of column names used as input keys.
+                input_keys, output_keys, metadata_keys must be disjoint, and must
+                exist in CSV column headers.
+            output_keys (Iterable[str]): List of column names used as output keys.
+                input_keys, output_keys, metadata_keys must be disjoint, and must
+                exist in CSV column headers.
+            metadata_keys (Iterable[str]): List of column names used as metadata keys.
+                input_keys, output_keys, metadata_keys must be disjoint, and must
+                exist in CSV column headers.
+            dataset_description: (Optional[str]): Description of the dataset.
+
+        Returns:
+            A Dataset object with the uploaded examples.
+        """
+        return self._upload_tabular_dataset(
+            table,
+            dataset_name=dataset_name,
+            input_keys=input_keys,
+            output_keys=output_keys,
+            metadata_keys=metadata_keys,
+            dataset_description=dataset_description,
         )
-        response.raise_for_status()
-        data = response.json()["data"]
-        version_id = data["version_id"]
-        examples = data["examples"]
-        print(f"💾 Examples uploaded: {self.web_url}datasets/{dataset_id}/examples")
-        print(f"🗄️ Dataset version ID: {version_id}")
 
-        return Dataset(
-            id=dataset_id,
-            version_id=version_id,
-            examples=[
-                Example(
-                    id=example["id"],
-                    input=example["input"],
-                    output=example["output"],
-                    metadata=example["metadata"],
-                    updated_at=datetime.fromisoformat(example["updated_at"]),
-                )
-                for example in examples
-            ],
+    def append_dataset(
+        self,
+        table: Union[str, Path, pd.DataFrame],
+        /,
+        *,
+        dataset_name: str,
+        input_keys: Iterable[str],
+        output_keys: Iterable[str] = (),
+        metadata_keys: Iterable[str] = (),
+    ) -> Dataset:
+        """
+        Append examples to dataset on the Phoenix server. If dataset does
+        not exist, it'll be created.
+
+        Args:
+            table (str | Path | pd.DataFrame): Location of a CSV text file, or
```

Comment:
I guess I don't know of a better param name but initially was slightly confused by this param name since I'm used to it being a dataframe.